### PR TITLE
Add az image galleries cleanup

### PR DESCRIPTION
**Pull request by narmaku**
This PR contains multiple commits. I recommend checking them one by one for better readability and understanding.

The main change is about restructuring azure.sh script to use functions and adding a new one that is in charge of cleaning up image galleries.

### CODE REVIEW

File: azure.sh
Comment (ondrejbudai) [inline azure.sh:121]: I think that the function should reset IFS to the original value when it returns.

**Review by ondrejbudai — COMMENTED** [review]
Looks good, just one minor nitpick.

File: azure.sh
Comment (narmaku) [reply azure.sh:122]: Makes sense! Done.

**Review by ondrejbudai — APPROVED** [review]
LGTM. @atodorov do you want to run a test before merging?

**Comment by atodorov** [comment]
> LGTM. @atodorov do you want to run a test before merging?

Yes. Scheduled here: https://github.com/osbuild/cloud-cleaner/actions/runs/6094607234

@narmaku FTR please execute the CI job yourself next time so that we can see it works without issues and avoid unnecessary delays.

File: oci.sh
Comment (atodorov) [inline oci.sh:50]: +1 for adding double quotes around the `$DRY_RUN` variable. 

However why change the double quotes around `true` with single quotes ? 

This also deviates from the pattern used elsewhere in the source code, for example here:
https://github.com/osbuild/cloud-cleaner/pull/31/files#diff-13c5717db0bbbc3eb9e1afbb8113d2bcc03a7aab9dedb451e2d35b93ebe91fdfR74

File: lib/common.sh
Comment (atodorov) [inline lib/common.sh:21]: IIRC in bash we'd prefer double quotes. Is there any particular reason to change them into single quotes?

**Review by atodorov — COMMENTED** [review]
Added comments. I will follow up with a PR to enable shellcheck here b/c it seems it isn't enabled.

**Comment by atodorov** [comment]
> > LGTM. @atodorov do you want to run a test before merging?
> 
> Yes. Scheduled here: https://github.com/osbuild/cloud-cleaner/actions/runs/6094607234
> 
> @narmaku FTR please execute the CI job yourself next time so that we can see it works without issues and avoid unnecessary delays.

The newly added code resulted in an error:

```
Cleaning image galleries...
--------------------
Checking civ_gallery_12245 gallery
  Checking civ-centos-9-x64-14597 image definition
    Checking 0.0.1 image version
Deleted image version 0.0.1
Error: Process completed with exit code 1.
```

File: lib/common.sh
Comment (narmaku) [reply lib/common.sh:21]: There's no particular reason, just a preference. I usually use double quotes only if needed, but if we agree on using double quotes everywhere that's totally fine for me.

Comment (atodorov) [reply lib/common.sh:21]: OK got it. 

I think ShellCheck will eliminate the question around preferences. Whatever it says is good that's what we use.

Comment (narmaku) [reply lib/common.sh:21]: I think ShellCheck doesn't complain about single vs. double quotes. I ran it locally and it didn't complain at all. So I think it's a matter of defining it as code style convention for this project (we can add it to README.md).

I just found this page that summarizes the differences quite well so that we can better decide:
* https://phoenixnap.com/kb/bash-single-vs-double-quotes

So, sometimes either one or the other is strictly needed, depending on what you want to do.

I think we can either agree on a sort of code style convention, or just allow both of them, based on developer's preference. (I prefer the first option but I understand it can be annoying sometimes).

Comment (ondrejbudai) [reply lib/common.sh:21]: We can add `shfmt` in a separate PR to end all style discussions once and forever. :D

https://github.com/mvdan/sh

Comment (narmaku) [reply lib/common.sh:21]: Oh yes, this is a great idea!

**Review by atodorov — CHANGES_REQUESTED** [review]
@narmaku 2 items:

1) Can we see a successful job execution of these changes before moving forward with merging ? 

2) Can you separate out the house-keeping changes (quotes, common functions, etc) into a separate PR? It looks like the first 2, maybe 3 commits aren't directly related to image galleries.

**Comment by jrusz** [comment]
> > Is this a blocker/no-go for this PR? If not, then I propose to keep it like this and I will take this into consideration for future PRs in this project. Thanks.
> 
> It's not a blocker per se, however given the PR is still in progress and that it's a matter of cherry-picking couple of commits IMO we can do it anyway. I'll leave others to jump in with their opinions.

IMO if it's all in separate commits it's fine and I wouldn't care about this being together in one PR. Where I think it would make sense to separate is if this was causing issues and was blocking this from merge repeatedly and this PR was opened for too long because of that but I don't think that's the case here. So if all is green I'd just keep it as is.

I do see your point though, it would be easy to cherry pick to a separate PR and would make it more clean and consistent I guess and easier for review, but since this is just a helper tool for our team I don't think we need to be so strict here.

**Comment by narmaku** [comment]
@atodorov , based on our last discussion on Monday, changes are ready to be review (if you like, you can only focus on the last commit where I made the latest changes). All the old galleries and their nested resources were cleaned up after three rounds of running, as expected (since there are three levels of nested resources). Good news is that many of them got fully deleted in the first iteration.
More information here:
https://github.com/osbuild/cloud-cleaner/actions/runs/6429579095/job/17463947452
(see the 3 attempts in the drop-down list right to the "Re-run jobs" button)

NOTE: The oci.sh job is failing but it's something that happens in main too. Not related to this PR.

**Pull request merged**
